custom/icds_reports/router.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import absolute_import, unicode_literals
+
+from corehq.sql_db.connections import connection_manager
+from custom.icds_reports.const import ICDS_UCR_ENGINE_ID
+
+ICDS_REPORTS_APP = 'icds_reports'
+
+
+class ICDSReportsRouter(object):
+
+    def db_for_read(self, model, **hints):
+        return db_for_read_write(model, write=False)
+
+    def db_for_write(self, model, **hints):
+        return db_for_read_write(model, write=True)
+
+    def allow_migrate(self, db, app_label, model=None, **hints):
+        if app_label != ICDS_REPORTS_APP:
+            return False
```

Comment:
I think this will prevent all migrations from other apps. I think it should be this:
```python
db_alias = get_icds_ucr_db_alias()
is_icds_app = app_label == ICDS_REPORTS_APP
if not db_alias and is_icds_app:
    return False
if db_alias == db:
    return is_icds_app

return None  # allow other routers to give input
```

Reply:
I think that https://github.com/dimagi/commcare-hq/pull/24084/commits/e7cb68d59c43656a25a7095648afc8f36e6eb232 is slightly simpler, with the same behavior